web/modules/custom/isa2_analytics/src/EventSubscriber/WebtoolsAnalyticsSubscriber.php

```diff
@@ -48,6 +59,24 @@ public function setSiteSection(AnalyticsEventInterface $event) {
 
     // Set the group label as the site section if there is an active group.
     if ($group = $this->ogContext->getGroup()) {
+      if ($group->bundle() === 'solution') {
+
+        $collection = NULL;
+        // Normally, the solution has the `collections` computed field that
+        // holds the list of collections that are affiliates to the solution.
+        // However, this method is fired during the preparation of the page
+        // while the computation of the value takes place during the rendering
+        // of the page.
+        // @see \Drupal\Core\Render\MainContent\HtmlRenderer::renderResponse
+        if ($collections = solution_get_collection_ids($group)) {
+          $collection_id = reset($collections);
```

Comment:
Let's put an explanation here why we are only taking the first collection. It might not be obvious to future generations that this is because of a business decision.